packages/x-date-pickers/src/DesktopTimePicker/DesktopTimePicker.tsx

```diff
@@ -23,25 +28,47 @@ const DesktopTimePicker = React.forwardRef(function DesktopTimePicker<TDate>(
   const localeText = useLocaleText<TDate>();
 
   // Props with the default values common to all time pickers
-  const defaultizedProps = useTimePickerDefaultizedProps<TDate, DesktopTimePickerProps<TDate>>(
-    inProps,
-    'MuiDesktopTimePicker',
-  );
+  const defaultizedProps = useTimePickerDefaultizedProps<
+    TDate,
+    TimeViewWithMeridiem,
+    DesktopTimePickerProps<TDate>
+  >(inProps, 'MuiDesktopTimePicker');
 
-  const viewRenderers: PickerViewRendererLookup<TDate | null, TimeView, any, {}> = {
-    hours: null,
-    minutes: null,
-    seconds: null,
+  const thresholdToRenderTimeInASingleColumn =
+    defaultizedProps.thresholdToRenderTimeInASingleColumn ?? 24;
+  const timeSteps = { hours: 1, minutes: 5, seconds: 5, ...defaultizedProps.timeSteps };
+  const shouldRenderTimeInASingleColumn =
+    (24 * 60) / timeSteps.minutes <= thresholdToRenderTimeInASingleColumn;
+
+  const renderTimeView = shouldRenderTimeInASingleColumn
+    ? renderDigitalClockTimeView
+    : renderMultiSectionDigitalClockTimeView;
+
+  const viewRenderers: PickerViewRendererLookup<TDate | null, TimeViewWithMeridiem, any, {}> = {
+    hours: renderTimeView,
+    minutes: renderTimeView,
+    seconds: renderTimeView,
+    meridiem: renderTimeView,
     ...defaultizedProps.viewRenderers,
   };
 
   const ampmInClock = defaultizedProps.ampmInClock ?? true;
+  const actionBarActions: PickersActionBarAction[] = !shouldRenderTimeInASingleColumn
+    ? ['accept']
+    : [];
```

Comment:
Seems easier to read without having to think to the negation
```suggestion
  const actionBarActions: PickersActionBarAction[] = shouldRenderTimeInASingleColumn
    ? []
    : ['accept'];
```